### PR TITLE
DM-40007: Update deprecation policy per RFC-945.

### PR DESCRIPTION
**Pull request by TallJimbo**

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -319,21 +319,14 @@ Documentation to be collected for the release notes in pipelines_lsst_io_ is:
...
-future releases.
+Deprecated interfaces may be removed in the major release after the one in which their deprecations first appear.
+These deprecations must be included in the release notes.
 
 To identify all deprecations that have to be mentioned in a release note, we
 search the codebase looking for specific strings. The application **ack** is
 used here as a reference, since it is easy to install in Unix systems [#ack]_.
 
-.. [#ack] The command **ack** may have to be installed separatelly.
+.. [#ack] The command **ack** may have to be installed separately.
```
Comment (ktlim) [inline index.rst:329]: Maybe should use `rg` now instead of `ack`.

File: index.rst [hunk 2]
```diff
@@ -319,21 +319,14 @@ Documentation to be collected for the release notes in pipelines_lsst_io_ is:
...
-
-- **Pending Deprecations**: the methods and functions that are marked as
-deprecated and will be removed **after** the next major release is done. This
-implies they will still be available for some time.
-
-- **Actual Deprecations**: the methods and functions that will be removed
-**before** the next major release. This implies they will not be available in
-future releases.
+Deprecated interfaces may be removed in the major release after the one in which their deprecations first appear.
+These deprecations must be included in the release notes.
```
Comment (ktlim) [inline index.rst:323]: While we won't have "actual deprecations" as defined previously, I think it's still good to have separate lists for the deprecations (deprecation warning in the code, may be removed in the next release) as well as the removals (code removed in this release) in the release notes.

**Review by ktlim — APPROVED** [review]

File: index.rst (hunk 1 above)
Comment (TallJimbo) [reply index.rst:329]: I took a brief stab at this, but I'm not a regular `rg` user and I didn't spot drop-in replacements for `ack`'s `--python` and `--cpp` options, so I'm going to leave it for now and let an expert make the change in the future.

File: index.rst (hunk 2 above)
Comment (TallJimbo) [reply index.rst:323]: I was surprised not to see a "removals" section mentioned in the previous text.  I'll add that.

**Pull request merged**